packages/filesystem/src/browser/filesystem-saveable-service.ts

```diff
@@ -89,26 +89,20 @@ export class FilesystemSaveableService extends SaveableService {
      * @param target The new URI for the widget.
      * @param overwrite
      */
-    private async copyAndSave(sourceWidget: Widget & SaveableSource & Navigatable, target: URI, overwrite: boolean): Promise<void> {
-        const snapshot = sourceWidget.saveable.createSnapshot!();
-        if (!await this.fileService.exists(target)) {
-            const sourceUri = sourceWidget.getResourceUri()!;
-            if (this.fileService.canHandleResource(sourceUri)) {
-                await this.fileService.copy(sourceUri, target, { overwrite });
-            } else {
-                await this.fileService.createFile(target);
-            }
-        }
-        const targetWidget = await open(this.openerService, target, { widgetOptions: { ref: sourceWidget } });
-        const targetSaveable = Saveable.get(targetWidget);
-        if (targetWidget && targetSaveable && targetSaveable.applySnapshot) {
-            targetSaveable.applySnapshot(snapshot);
-            await sourceWidget.saveable.revert!();
-            sourceWidget.close();
-            Saveable.save(targetWidget, { formatType: FormatType.ON });
+    protected async copyAndSave(sourceWidget: Widget & SaveableSource & Navigatable, target: URI, overwrite: boolean): Promise<void> {
+        const saveable = sourceWidget.saveable;
+        const snapshot = saveable.createSnapshot!();
+        const content = Saveable.Snapshot.read(snapshot) ?? '';
+        if (await this.fileService.exists(target)) {
+            // Do not fire the `onDidCreate` event as the file already exists.
+            await this.fileService.write(target, content);
         } else {
-            this.messageService.error(nls.localize('theia/workspace/failApply', 'Could not apply changes to new file'));
+            // Ensure to actually call `create` as that fires the `onDidCreate` event.
+            await this.fileService.create(target, content, { overwrite });
         }
+        await saveable.revert!();
+        sourceWidget.close();
+        await open(this.openerService, target, { widgetOptions: { ref: sourceWidget } });
```

Comment:
I know you didn't touch that line but I'm seeing the following error when saving a file as another:
```
Error: Reference widget is not in the layout.
    at __webpack_modules__.../../node_modules/@phosphor/widgets/lib/docklayout.js.DockLayout.addWidget (docklayout.js:326:1)
    at __webpack_modules__.../../node_modules/@phosphor/widgets/lib/dockpanel.js.DockPanel.addWidget (dockpanel.js:307:1)
    at TheiaDockPanel.addWidget (theia-dock-panel.ts:140:15)
    at ApplicationShellWithToolbarOverride.addWidget (application-shell.ts:938:32)
    at EditorPreviewManager.doOpen (editor-preview-manager.ts:76:24)
    at EditorPreviewManager.open (widget-open-handler.ts:90:20)
    at async FilesystemSaveableService.copyAndSave (filesystem-saveable-service.ts:107:9)
    at async FilesystemSaveableService.saveAs (filesystem-saveable-service.ts:81:17)
```
Basically preventing the new file from being opened:

![save-as](https://github.com/eclipse-theia/theia/assets/19170971/af03b477-5f3a-454f-8731-b6165c6dcdfe)

Do you see the same?

Reply:
I can reproduce, the latest changes are supposed to fix this.